src/activities/CategoriesEntity.js

```diff
@@ -89,6 +89,29 @@ export class CategoriesEntity extends Entity {
 		return Number(category.categoryId) === Number(selectedCategory.properties.categoryId);
 	}
 
+	getSelectedCategoryHref(categoryId) {
+		if (!this._entity) {
+			return;
+		}
+
+		const subEntities = this._entity.getSubEntitiesByClass(Classes.activities.category);
+
+		let selectedCategory;
+		if (subEntities) {
+			selectedCategory = subEntities.find(category => Number(category.properties.categoryId) === categoryId);
+		}
+
+		if (!selectedCategory) return;
+
+		let action = selectedCategory.getActionByName(Actions.activities.categories.deselect);
```

Comment:
If the user chooses the same category, this let's us get the href and set it for consistency.

Reply:
Like one that wasn’t marked as selected? I thought it would always be deselect…?

Reply:
`deselect` only appears once in the collection of categories (it indicates that category is the currently selected category at serialization when the GET is made, but not at present in the UI).

![image](https://user-images.githubusercontent.com/15062048/197965267-1bff31ad-2b9d-4e88-88e3-02505c54e6ab.png)

Reply:
The way we use it right now (in activities) persists the categories collection and refetches the collection so technically the selected category should always be updated I think, but of course it's always safer to do this! Just curious if you actually ran into issues because of this 👀